### PR TITLE
Updates to the model API and documentation

### PR DESCRIPTION
**Pull request by lshannon**
Hi @RichardW98 can you please review and update the workflow-example project to see if these changes make sense?

### CODE REVIEW

**Review by pkliczewski — COMMENTED** [review]
Few nitpicks.

I saw some code formatting differences. Have we decided to use common code formatter?

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+@Setter
+public class WorkFlowOption {
+	
+	private WorkFlowDefinition workFlowDefinition;
+	private String name;
+	private String description;
+	private List<String> details;
+	private String workFlowId;
+
+	private WorkFlowOption(Builder builder, List<String> details) {
```
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java:57]: @lshannon good catch

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		this.description = builder.workFlowDefinition.getDescription();
+		this.details = builder.details;
+		this.workFlowId = builder.workFlowDefinition.getId();
+		this.name = builder.workFlowDefinition.getName();
+		this.details = builder.details;
```
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java:62]: @pkliczewski good catch

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+	 *
+	 * @param otherOption new Workflow Option to add to the other options list
+	 * @return the updated reference
+	 */
+
+	public List<WorkFlowOption> getOtherOptions() {
+		return otherOptions;
+	}
+
+	public List<WorkFlowOption> addOtherOption(WorkFlowOption otherOption) {
```
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java:97]: @pkliczewski good catch

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/assessment/BaseAssessmentTask.java [hunk 4]
```diff
@@ -30,13 +32,13 @@ public abstract class BaseAssessmentTask implements WorkFlowTask {
 	/**
 	 * These are the options this AssessmentTasks can return
 	 */
-	InfrastructureOption infrastructureOptions;
+	List<WorkFlowOption> workflowOptions;
```
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/assessment/BaseAssessmentTask.java:35]: @pkliczewski will look into this

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	/**
+	 * Add a Continuation Options
+	 *
+	 */
+	public List<WorkFlowOption> getContinuationOptions() {
+		return continuationOptions;
+	}
+
+	public List<WorkFlowOption> addContinuationOption(WorkFlowOption continuationOption) {
+		upgradeOptions.add(continuationOption);
```
Comment (pkliczewski) [inline parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java:89]: Here we should use `continuationOptions`

**Review by pkliczewski — COMMENTED** [review]
A lot of formatting changes. It is really hard to understand what has changed.

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java (hunk 5 above)
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java]: This is addressed

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/assessment/BaseAssessmentTask.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/assessment/BaseAssessmentTask.java:36]: Made this private

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOptions.java:107]: Resolved

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java:66]: Resolved

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/option/WorkFlowOption.java]: Resolved

**Review by pkliczewski — COMMENTED** [review]
looks good to me

File: notification-service/src/main/java/com/redhat/parodos/notification/config/SwaggerConfiguration.java [hunk 6]
```diff
@@ -15,24 +15,25 @@
...
-    OpenAPI springShopOpenAPI() {
-        return new OpenAPI()
-                .info(new Info().title("Parodos Notification Service")
-                        .description("This is the API documentation for the Parodos Notification Service. " +
-                                "The APIs are secured with oAuth2/OpenID and cannot be accessed without a valid token.")
-                        .version("v1.0.0"));
-    }
+
+	@Bean
+	OpenAPI springShopOpenAPI() {
```
Comment (anludke) [inline notification-service/src/main/java/com/redhat/parodos/notification/config/SwaggerConfiguration.java:32]: rename to stg like _parodosOpenAPI_()

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/exception/MissingParameterException.java [hunk 7]
```diff
@@ -13,16 +13,20 @@
...
- * Thrown when the WorkContext is missing a WorkParameter required by a Task
- * 
+ * Thrown when the @see WorkContext is missing a @see WorkFlowTaskParameter
+ *
  * @author Luke Shannon (Github: lshannon)
  *
  */
 public class MissingParameterException extends Exception {
+
+	private static final long serialVersionUID = 1L;
```
Comment (anludke) [inline parodos-model-api/src/main/java/com/redhat/parodos/workflow/exception/MissingParameterException.java:26]: can be removed

File: notification-service/src/main/java/com/redhat/parodos/notification/config/SwaggerConfiguration.java (hunk 6 above)
Comment (lshannon) [reply notification-service/src/main/java/com/redhat/parodos/notification/config/SwaggerConfiguration.java:32]: Done

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/exception/MissingParameterException.java (hunk 7 above)
Comment (lshannon) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/exception/MissingParameterException.java:26]: Its unlikely that someone would serialize an exception. But, I personally like to set the serialVersionUID to 1L on objects so in the event the object model changes, and someone has a an older version of the object serialized (ie: an in-memory cache would do this) reading the stored object value into the new version of the object will not result in an serialization error.  On the flip side, we ensure our objects are backwards compatible.

**Review by anludke — APPROVED** [review]

**Pull request merged**